Week-5/session-15/ maps and sets/index.js

```diff
@@ -0,0 +1,27 @@
+let productViews = new WeakMap();
+let cartItems = new WeakSet();
+
+function incrementProductViews(productId) {
+  if (productViews.has(productId)) {
+    productViews.set(productId, productViews.get(productId) + 1);
+  } else {
+    productViews.set(productId, 1);
+  }
+}
+
+function addToCart(productId) {
+  if (cartItems.has(productId)) {
+    return "Product already in cart";
+  } else {
+    cartItems.add(productId);
+    return "Product added to cart";
+  }
+}
+
+incrementProductViews(123);
```

Comment:
WeakMap and WeakSet are used in specific cases where the keys are objects, this is invalid it can be `let product1 = { id: 123 };`